erpnext/stock/doctype/quality_inspection/quality_inspection.py

```
@@ -41,6 +47,9 @@ def get_quality_inspection_template(self):
 
 	def on_submit(self):
 		self.update_qc_reference()
+		self.validate_certificate_of_analysis()
```

This will run after the document is submitted, we should throw this error first. Do this on `before_submit` instead.